### ISSUE TITLE
Include org data as a column behind suffixes in the public suffix dataset

### ISSUE THREAD

**Issue by soodoku** [Ironholds/urltools#40]
Dear Oliver,

First, congratulations on a fantastic library --- it is fast and clean. 

My request is pretty straight forward --- and it may well be that I am missing something: 

the publicsuffixlist carries a bunch of data in the comments:
https://publicsuffix.org/list/public_suffix_list.dat

It would be awesome to incorporate that data. It is easily done. And perhaps already is. If not, it would be great to see it.

**Comment by Ironholds** [comment]
It is! https://github.com/Ironholds/urltools/blob/master/R/suffix.R

**Comment by soodoku** [comment]
Dear Oliver,

I apologize but I meant simply that it would be great to have col. listing which org. is behind the suffix. So for instance, when we pick out: at-band-camp.net, we also list dyndns.com?

https://publicsuffix.org/list/public_suffix_list.dat

**Comment by Ironholds** [comment]
Ohh, gotcha. Makes sense!

**Comment by okeyes-r7** [comment]
@jayjacobs wanna take a look at this when you get some free time? You know more about munging and parsing this dataset than me.

**Comment by jayjacobs** [comment]
next few weeks are going to be crazy, but this should be doable.

**Comment by Ironholds** [comment]
Jay, any luck on this?

**Comment by jayjacobs** [comment]
I'm giving up on this, you've moved all the suffix code out of R.  While I could attempt this change, I'm quite sure you'd wish I hadn't. 

I won't close this however because it's still not a bad addition.

**Comment by Ironholds** [comment]
Oh, possible miscommunication; not looking at incorporating the dataset
into the results of suffix_extract, just into the suffixes dataset itself.
So it's the suffix updating code that needs tweaking, and that should still
be in R.

**Comment by jayjacobs** [comment]
Gotcha, given that the suffixes dataset is just a vector, the best we could do without messing up suffix_extract is to create a named vector, where the name is the comments that apply to the suffixes listed.  I can get that in there...

**Comment by Ironholds** [comment]
Or I'll just slightly tweak the logic of the suffix extractor to grab the
right vector!

**Comment by jayjacobs** [comment]
what if the suffix dataset were changed to a data.frame?  That way we could have a column for the comments around a suffix and then the suffix itself.  suffix_extract would then just create a vector from the data.frame.  I think that could all be in R, because it'll pass off the same vector in the suffix_extract function.

**Comment by Ironholds** [comment]
Exactly what I was thinking.

**Comment by jayjacobs** [comment]
on it.

**Comment by jayjacobs** [comment]
Here's the logic I'm going with, because the comments are totally non-standard.  

1. First, I made sure whatever I did didn't drop or miss any suffixes.  
2. Then I break the file up on blank lines (blank lines are not enforced in the format of the file, but it looks to be commonly done. 
3. Next I separate all the comments in that section
4. and use the first comment to represent the entire section. 

In my first pass, I put ALL the comments in the section  the comments, but that got crazy.  I also experimented with tagging some of the comments lower on (like in the .us when they have a "regional" comment).  But none of that worked because comments are also used to remove valid suffixes and those aren't descriptive comments at all. 

Also, I am modifying suffix_extract to accept either the old vector style or a new data.frame style that suffix_extract is taking.
